src/foam/u2/view/FObjectView.js

```diff
@@ -88,12 +88,18 @@ foam.CLASS({
     },
     {
       class: 'Object',
-      name: 'persistantData',
-      documentation: 'Allows passed in data to persist to selected class instances.'
+      name: 'persistentData',
+      documentation: `
+        Allows passed in data to persist to selected class instances.
```

Comment:
I still don't understand. Why don't you just copy over all parameters?

Reply:
default values and factories of the new class won't be applied by copying.

Reply:
Couldn't you just walk all of the axioms and copy them yourself to avoid this problem?

        List<PropertyInfo> props = getClassInfo().getAxiomsByClass(PropertyInfo.class);
        for ( PropertyInfo p : props ) {
          try {
            if ( p.isSet(obj) ) p.set(this, p.get(obj));
          } catch (ClassCastException e) {
            try {
              PropertyInfo p2 = (PropertyInfo) obj.getClassInfo().getAxiomByName(p.getName());
              if ( p2 != null ) {
                p.set(this, p2.get(obj));
              }
            } catch (ClassCastException e2) {}
          }
        }

Reply:
java? this is client side code.

Reply:
Then even simpler on the client, just use obj.copyFrom(obj2).